app/controllers/user_credentials_controller.rb

```diff
@@ -14,6 +14,24 @@ def index
     end
   end
 
+  def import
+    file = params[:file]
+    content = JSON.parse(file.read)
+    new_credentials = content.map do |hash|
+      current_user.user_credentials.build(hash.slice("credential_name", "credential_value", "mode"))
+    end
+
+    if new_credentials.map(&:save).all? { |save_status| save_status }
+      respond_to do |format|
+        format.html { redirect_to user_credentials_path, notice: "The file was successfully uploaded."}
+      end
+    else
+      respond_to do |format|
+        format.html { redirect_to user_credentials_path, notice: 'One or more of the uploaded was not imported due to duplicate credential name error', status: 422 }
```

Comment:
I added , `status: 422`. I am having a redirection error locally but I am aware that this is for the server so I can't test it.